### PR TITLE
Add Gitlab Plugin boilerplate code

### PR DESCRIPTION
**Pull request by tigrato**
This PR introduces the proto and compiled proto changes required to implement an Access Graph Gitlab plugin.

Part of https://github.com/gravitational/access-graph/issues/635

### CODE REVIEW

File: api/types/plugin.go [hunk 1]
```diff
@@ -76,6 +78,8 @@ const (
 	PluginSubkindMDM = "mdm"
 	// PluginSubkindAccess represents access request plugins collectively
 	PluginSubkindAccess = "access"
+	// PluginSubkindAccessGraph represents access graph plugins collectively
+	PluginSubkindAccessGraph = "accessgraph"
```
Comment (bl-nero) [inline api/types/plugin.go:82]: How is this relevant to this PR?

**Review by bl-nero — APPROVED** [review]

File: api/types/plugin.go [hunk 2]
```diff
@@ -596,3 +613,12 @@ func (c *PluginOAuth2AccessTokenCredentials) CheckAndSetDefaults() error {
 func (c PluginStatusV1) GetCode() PluginStatusCode {
 	return c.Code
 }
+
+// CheckAndSetDefaults checks that the required fields for the Gitlab plugin are set.
+func (c *PluginGitlabSettings) Validate() error {
```
Comment (justinas) [inline api/types/plugin.go:618]: `CheckAndSetDefaults()`, unless we're migrating away from that convention? 🙂

**Review by justinas — APPROVED** [review]

File: api/types/plugin.go (hunk 2 above)
Comment (tigrato) [reply api/types/plugin.go:618]: We are moving away from CheckAndSetDefaults

File: api/types/plugin.go (hunk 1 above)
Comment (tigrato) [reply api/types/plugin.go:82]: This works as a plugin subtype to easily identify which plugins are associated with access graph. It's used by the enterprise PR

**Review by jakule — APPROVED** [review]
